src/Core/src/Core/MauiApp.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Maui
+{
+	public sealed class MauiApp : IHost
+	{
+		private readonly IHost _host;
+
+		internal MauiApp(IHost host)
+		{
+			_host = host;
+			Logger = host.Services.GetRequiredService<ILoggerFactory>().CreateLogger("TODO: Environment.ApplicationName");
+		}
+
+		/// <summary>
+		/// The application's configured services.
+		/// </summary>
+		public IServiceProvider Services => _host.Services;
+
+		/// <summary>
+		/// The application's configured <see cref="IConfiguration"/>.
+		/// </summary>
+		public IConfiguration Configuration => _host.Services.GetRequiredService<IConfiguration>();
+
+		/// <summary>
+		/// Allows consumers to be notified of application lifetime events.
+		/// </summary>
+		public IHostApplicationLifetime Lifetime => _host.Services.GetRequiredService<IHostApplicationLifetime>();
+
+		/// <summary>
+		/// The default logger for the application.
+		/// </summary>
+		public ILogger Logger { get; }
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="MauiApp"/> class with preconfigured defaults.
+		/// </summary>
+		/// <returns>The <see cref="MauiApp"/>.</returns>
+		public static MauiApp Create() => new MauiAppBuilder().Build();
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="MauiAppBuilder"/> class with preconfigured defaults.
+		/// </summary>
+		/// <returns>The <see cref="MauiAppBuilder"/>.</returns>
+		public static MauiAppBuilder CreateBuilder() => new();
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="MauiAppBuilder"/> class with optional defaults.
+		/// </summary>
+		/// <param name="useDefaults">Whether to create the <see cref="MauiAppBuilder"/> with common defaults.</param>
+		/// <returns>The <see cref="MauiAppBuilder"/>.</returns>
+		public static MauiAppBuilder CreateBuilder(bool useDefaults = true) => new(useDefaults);
+
+		/// <summary>
+		/// Start the application.
+		/// </summary>
+		/// <param name="cancellationToken"></param>
+		/// <returns>
+		/// A <see cref="Task"/> that represents the startup of the <see cref="MauiApp"/>.
+		/// Successful completion indicates the HTTP server is ready to accept new requests.
+		/// </returns>
+		public Task StartAsync(CancellationToken cancellationToken = default) =>
```

Comment:
Start and stop should probably be implemented explicitly because I don't think they are used in maui

Reply:
I think it could be nice to fully support IApplicationLifetime et al so I'll look into what we can do here.

Reply:
Yeah, _we_ may not use it for maui, but we still want Bob's library to work.